experiments/pipeline/examples/general_usage.md

```diff
@@ -0,0 +1,195 @@
+
+In case **there are more schemas in the pipeline**, the data will be loaded into dataset with name `{dataset_name}` for default schema and `{dataset_name}_{schema_name}` for all the other schemas. For the example above:
+1. `spotify` tables and `labels` will load into `spotify_data_1`
+2. `mel` resource will load into `spotify_data_1_echonest`
+
+
+## pipeline working directory and state
+Another fundamental concept is the pipeline working directory. This directory keeps the following information:
+1. the extracted data and the load packages with jobs created by normalize
+2. the current schemas with all the recent updates
+3. the pipeline and source state files.
+
+**Pipeline working directory should be preserved between the runs - if possible**
+
+If the working directory is not preserved:
+1. the auto-evolved schema is reset to the initial one. the schema evolution is deterministic so it should not be a problem - just a time wasted to compare schemas with each run
+2. if load package is not fully loaded and erased then the destination holds partially loaded and not committed `load_id`
+3. the sources that need source state will not load incrementally.
+
+This is the situation right now. We could restore working directory from the destination (both schemas and state). Entirely doable (for some destinations) but can't be done right now.
+
+## running pipelines and `dlt.run` + `@source().run` functions
+`dlt.run` + `@source().run` are shortcuts to `Pipeline::run` method on default or last configured (with `dlt.pipeline`) `Pipeline` object. Please refer to [create_pipeline.md](create_pipeline.md) for examples.
+
+The function takes the following parameters
+1. source - required - the data to be loaded into destination: a `@dlt.source` or a list of those, a `@dlt.resource` or a list of those, an iterator/generator function or a list of those or iterable (ie. a list) holding something else that iterators.
+2. destination
+3. dataset name
+4. table_name, write_disposition etc. - only when data is: a single resource, an iterator (ie. generator function) or iterable (ie. list)
+5. schema - a `Schema` instance to be used instead of schema provided by the source or the default schema
+
+The `run` function works as follows.
+1. if there's any pending data to be normalized or loaded, this is done first.
+2. only when successful more data is extracted
+3. only when successful newly extracted data is normalized and loaded.
+
+extract / normalize / load are atomic. the `run` is as close to be atomic as possible.
+
+the `run` and `load` return information on loaded packages: to which datasets, list of jobs etc. let me think what should be the content
+
+> `load` is atomic if SQL transformations ie in `dbt` and all the SQL queries take into account only committed `load_ids`. It is certainly possible - we did in for RASA but requires some work... Maybe we implement a fully atomic staging at some point in the loader.
```

Comment:
atomic staging is the way to go IMO

Reply:
the "load id" thing for rasa is useful for incremental processing but can be implemented without our help